exopy_hqc_legacy/instruments/drivers/visa/agilent_pna.py

```diff
@@ -80,14 +80,26 @@ def read_formatted_data(self, meas_name=''):
             self.selected_measure = meas_name
         else:
             meas_name = self.selected_measure
+            
+        # get the endianess of the output so that the data is read properly
+        data_endianess = self._pna.data_endianess
+        if data_endianess == 'little':
+            is_big_endian = False
+        elif data_endianess == 'big':
+            is_big_endian = True
+        else:
+            raise InstrIOError(cleandoc('''Agilent PNA did not define 
+                                        endianess properly'''))
 
         data_request = 'CALCulate{}:DATA? FDATA'.format(self._channel)
-        if self._pna.data_format == 'REAL,32':
-            data = self._pna.query_binary_values(data_request, 'f')
-
+        if self._pna.data_format in ('REAL,32', 'REAL,+32'):
+            data = self._pna.query_binary_values(data_request, datatype='f', 
```

Comment:
All those could be made more efficient by passing `container=np.ndarray` to directly obtain a numpy array rather than a list. This also apply to the ascii transfer